### PR TITLE
Python SDK v3

### PR DESCRIPTION
**Pull request by nghiant2710**
This PR contains some fixes for API v3. We need to release this before adding API v4 (multicontainer) support.

Most of the commits are reviewed, only the last one is added recently.

### CODE REVIEW

File: resin/models/build.py [hunk 1]
```diff
@@ -43,6 +43,25 @@ def get(self, id):
         else:
             raise exceptions.BuildNotFound(id)
 
+    def get_by_commit(self, commit_hash):
+        """
+        Get a specific build by commit hash.
```
Comment (pimterry) [inline resin/models/build.py]: I think the wording isn't quite right: there can easily be multiple builds for a commit hash (just only one successful build), so this should be something like `Get all builds for a commit hash`

File: resin/models/device.py [hunk 2]
```diff
@@ -692,34 +695,46 @@ def disable_device_url(self, uuid):
...
 
         Args:
             uuid (str): device uuid.
-            build (str): build id.
+            build_commit_hash (str): build commit hash.
 
         Raises:
             DeviceNotFound: if device couldn't be found.
             ApplicationNotFound: if application couldn't be found.
+            FailedBuild: if build commit hash points to an error build.
```
Comment (pimterry) [inline resin/models/device.py]: For similar reasons, this should be `if all builds for the commit hash failed.`

**Review by pimterry — CHANGES_REQUESTED** [review]

File: resin/models/build.py (hunk 1 above)
Comment (nghiant2710) [reply resin/models/build.py]: @pimterry updated!

File: resin/models/device.py (hunk 2 above)
Comment (nghiant2710) [reply resin/models/device.py]: Updated!

**Review by pimterry — APPROVED** [review]

**Pull request merged**